src/CTA.Rules.Analysis/RulesAnalysis.cs

```diff
@@ -201,7 +201,7 @@ private bool AnalyzeChildren(FileActions fileAction, UstList<UstNode> children,
                                 }
 
                                 token = null;
-                                foreach (string interfaceName in classType.BaseList)
+                                foreach (string interfaceName in classType.BaseList ?? Enumerable.Empty<string>())
```

Comment:
why Enumerable.Empty<string> ? we can skip this block if the classType.BaseList is empty?

Reply:
piggyback on this, could you add some comment explaining the change inline? In what situation this could be null for example

Reply:
Using Enumerable.Empty will result in the block being skipped. I just felt it was more concise like this than wrapping the whole block in an if statement.

ClassDeclaration is a Codelyzer class, after looking at the code for a while it seems like the only explanation is that sometimes when a class has an empty base list, rather than having an empty collection here, the value is just null. Will add a comment explaining this.

Reply:
Thanks for the update. Could you elaborate a bit on the logic here as well since this change will only skip the foreach() section, but would continue with the next AnalyzeChildren attempt. Is that expected or we should break right after we know the classType.BaseList is null?